vike-react/renderer/+config.ts

```diff
@@ -44,6 +74,10 @@ export default {
     },
     lang: {
       env: 'server-only'
+    },
+    ssr: {
+      env: 'server-and-client',
```

Comment:
I'm now getting

```txt
[vite-plugin-ssr][config][Wrong Usage] vike-react/vike-react/dist/renderer/+config.js > meta.ssr.effect is only supported if meta.ssr.env is 'config-only' (but it's 'server-and-client')
```

But we don't want `config-only` here, correct?

Reply:
oh I see, I could use a `config-only` flag that in turns toggles another flag with `env: server-and-client`. Is this also what you would do?

Reply:
Yes that's what I had in mind: `ssr` is `config-only` while its `effect` sets `Page` to `server-and-client`/`client-only`.

Reply:
Then, to discriminate  in the `onRender*` hooks, we check whether `Page` is defined. That way, I believe we can implement it like `examples/render-modes/`.

Reply:
I've just had a look at https://github.com/brillout/vite-plugin-ssr/tree/main/examples/render-modes-v1 - I get it now, thanks. Let me rework this PR